src/internal/direction/index.ts

```diff
@@ -0,0 +1,74 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { PointerEvent as ReactPointerEvent } from 'react';
+
+export function getIsRtl(element: null | HTMLElement | SVGElement): boolean {
+  if (!element) {
+    return false;
+  }
+  return getComputedStyle(element).direction === 'rtl';
+}
+
+export function getOffsetInlineStart(element: HTMLElement) {
+  const offsetParentWidth = element.offsetParent?.clientWidth ?? 0;
+  return getIsRtl(element) ? offsetParentWidth - element.offsetWidth - element.offsetLeft : element.offsetLeft;
+}
+
+/**
+ * The scrollLeft value will be a negative number if the direction is RTL and
+ * needs to be converted to a positive value for direction independent scroll
+ * computations. Additionally, the scrollLeft value can be a decimal value on
+ * systems using display scaling requiring the floor and ceiling calls.
+ */
+export function getScrollInlineStart(element: HTMLElement) {
+  return getIsRtl(element) ? Math.floor(element.scrollLeft) * -1 : Math.ceil(element.scrollLeft);
+}
+
+/**
+ * The clientX position needs to be converted so it is relative to the right of
```

Comment:
"right" implies that it's _always_ the right side, maybe we can phrase this as being the inline-start side? (same for the description of `getLogicalPageX` below)